File: RayTracingInCPP/Vector3.cpp
//
// Created by Abdulaziz on 1/15/2025.
//

#include "Vector3.h"


Vector3::Vector3(const double x, const double y, const double z)
    : x(x), y(y), z(z) {}

double Vector3::dot(const Vector3& vector3) const {
    return x * vector3.x +y * vector3.y + z * vector3.z;
}

Vector3 Vector3::Subtract(const Vector3& vector3) const {
    return Vector3(x - vector3.x, y - vector3.y, z - vector3.z);
}

double Vector3::getX() const {
    return x;
}

double Vector3::getY() const {
   return y;
}

double Vector3::getZ() const {
   return z;
}

File: RaytracerJava/phase1/src/Vector3.java
public class Vector3 {
    private final double x;
    private final double y;
    private final double z;

    public Vector3(double x,double y,double z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    public Vector3 subtract(Vector3 vector){
        return new Vector3(x-vector.getX(),y-vector.getY(),z-vector.getZ());
    }

    public double dot(Vector3 vector) {
        return x * vector.x + y * vector.y + z * vector.z;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public double getZ() {
        return z;
    }
}
